src/role-helpers.js

```diff
@@ -134,7 +134,13 @@ function getRoles(container, {hidden = false} = {}) {
       return hidden === false ? isInaccessible(element) === false : true
     })
     .reduce((acc, node) => {
-      const roles = getImplicitAriaRoles(node)
+      let roles = []
+      // TODO: This violates html-aria which does not allow any role on every element
+      if (node.hasAttribute('role')) {
+        roles = node.getAttribute('role').split(' ').slice(0, 1)
```

Comment:
Can an element have multiple roles? If so, is there a reason we're only getting the first one here?

Reply:
An element can only have a single role. We only consider arrays right now for historic reasons. At some point aria-query will return a single role but this requires more work that isn't actually blocking this PR. 

If you pass in multiple roles then those are considered fallback roles. We allow querying them but haven't decided yet if we want to include it in the error message. It gets quite complicate since you have to branch one more time. 

I don't think this is a concern of this PR. aria-query returning multiple roles or `queryFallback` not being considered can be solved independently.